openml/tasks/task.py

```diff
@@ -17,6 +17,14 @@
 
 
 class OpenMLTask(ABC):
+    """OpenML Task object.
+
+       Parameters
+       ----------
+       task_type_id : int or str
```

Comment:
Seing this I'm surprised that we allow strings here and cast them to int inside the constructor. Could you please update the code so that:
1. the casting happens outside
2. the docstring reflects this
3. the type annotation reflects this, too

Reply:
By casting outside, are you referring to an explicit cast made by the user or in any function invoking the constructor? Just checked some other constructors, and they accept ids only as `int`. 
Shall I make the id parameters take only `int` here too?

Reply:
I mean 'any function invoking the constructor'. Yes, it would be good to only accepts integers here, too.